xbmc/cores/VideoPlayer/DVDCodecs/Overlay/contrib/cc_decoder708.cpp

```diff
@@ -172,21 +172,21 @@ int compWindowsPriorities (const void *a, const void *b)
 
 void clearTV (cc708_service_decoder *decoder)
 {
-  for (int i=0; i<I708_SCREENGRID_ROWS; i++)
+  for (auto& c : decoder->tv.chars)
   {
-    memset (decoder->tv.chars[i], ' ', I708_SCREENGRID_COLUMNS);
-    decoder->tv.chars[i][I708_SCREENGRID_COLUMNS]=0;
+    memset (c, ' ', I708_SCREENGRID_COLUMNS);
+    c[I708_SCREENGRID_COLUMNS]=0;
   }
 };
 
 void printTVtoBuf (cc708_service_decoder *decoder)
 {
   int empty=1;
   decoder->textlen = 0;
-  for (int i=0;i<75;i++)
+  for (auto& c : decoder->tv.chars)
```

Comment:
75 is not entire range

Reply:
`unsigned char chars[I708_SCREENGRID_ROWS][I708_SCREENGRID_COLUMNS+1];` and `#define I708_SCREENGRID_ROWS 75` so why should 75 not be the entire range?

Reply:
see above, not clear enough what we get from auto c